lib/git_service/commit.rb

```diff
@@ -23,21 +23,63 @@ def rugged_diff(other_ref = parent_oids.first)
       other_commit.diff(rugged_commit)
     end
 
+    def formatted_author
+      "#{rugged_commit.author[:name]} <#{rugged_commit.author[:email]}>"
+    end
+
+    def formatted_author_date
+      rugged_commit.author[:time].to_time.strftime("%c %z")
+    end
+
+    def formatted_committer
+      "#{rugged_commit.committer[:name]} <#{rugged_commit.committer[:email]}>"
+    end
+
+    def formatted_committer_date
+      rugged_commit.committer[:time].to_time.strftime("%c %z")
+    end
+
+    def formatted_commit_stats
+      diff.file_status.map do |file, stats|
+        file_stats  = file
+        file_stats << " | "
+        file_stats << (stats[:additions].to_i + stats[:deletions].to_i).to_s
+        file_stats << " "
+        file_stats << "+" if stats[:additions].positive?
+        file_stats << "-" if stats[:deletions].positive?
```

Comment:
I did in fact go down the rabbit hole of looking into this:

https://github.com/ManageIQ/miq_bot/pull/398#discussion_r162480938

And for the time being, decided to eject from trying to handle that at the moment (I was digging into the internals of `git` for a bit before I realized I wasn't finding what I was looking for).  Effectively, the formatting that is done is based on the `width` available in `git log --stat[=width]` (where `width` is optional, and defaults to the width of the current terminal), so a decent amount of considerations are made to balance out the fields, and the "column sizes" (file path, total changes, additions/deletions graph) can vary based on the contents of each.

A couple of interesting commits from this repo (viewed by running `git log --stat=80 [SHA]` highlight some of the different scenarios based on the width:

- 7258ed4fc755a8ce3eca3249902bc38475f075cc
- b7916ca5c25f1b2bfd5d423b2086aa2cd4d979b0
- b327fd06e6db992b2ec332e92e9327d239a50a73
- a6f54c26aa9d099ec6cf01702b5c9f784093a83d

If we decide to attempt that at some point, the way I have broken out the method here should make it a little more readable to do just that.

Reply:
Yeah, we noticed that in the original PR and also bailed on it, cause it wasn't really that important to be *exactly* like git in that respect.

Reply:
Yeah, mostly wanted to provide some context in this PR that "research was done", but ended up not going forward, since it wasn't completely clear why we bailed previously.